cloudferrylib/os/actions/check_needed_compute_resources.py

```diff
@@ -61,27 +61,20 @@ def check_in_use_flavor(self, objs, info):
         # is in this flavor, it will keep the old flavor id, can not be matched
         # to existing flavors
         src_compute = self.src_cloud.resources[utl.COMPUTE_RESOURCE]
-        src_flavor_ids = \
-            [flavor.id for flavor in src_compute.get_flavor_list()]
         dst_compute = self.dst_cloud.resources[utl.COMPUTE_RESOURCE]
-        dst_flavors = dst_compute.get_flavor_list()
-        for instance in objs.values():
-            inst_flavor_id = instance['instance']['flavor_id']
-            _instance = instance['instance']
-            instance_id = _instance['id']
-            flav_details = \
-                info['instances'][instance_id]['instance']['flav_details']
-            re_create_dst_flavor = False
-            for flavor in dst_flavors:
-                if flavor.name == flav_details['name']:
-                    dst_compute.delete_flavor(flavor.id)
-                    re_create_dst_flavor = True
-            if inst_flavor_id not in src_flavor_ids or re_create_dst_flavor:
-                dst_compute.create_flavor(name=flav_details['name'],
-                                          flavorid=_instance['flavor_id'],
-                                          ram=flav_details['memory_mb'],
-                                          vcpus=flav_details['vcpus'],
-                                          disk=flav_details['root_gb'],
-                                          ephemeral=flav_details[
-                                              'ephemeral_gb'])
-                src_flavor_ids.append(_instance['flavor_id'])
+        dst_flavors = {f.name: f for f in dst_compute.get_flavor_list()}
```

Comment:
I  think it would be better to simply remove whole `check_in_use_flavor` method, since it's functionality is already covered in `nova_compute.create_flavor_if_not_exists`